### DISCUSSION TITLE
resticker container just exits after restic is done

### DISCUSSION THREAD

**Discussion by vsuontam** [djmaze/resticker#135] [General]
Hi,

I am using resticker container in my Synology drive to back it to Backblaze. It is through the Docker Hub:
https://hub.docker.com/r/mazzolino/restic

It all works well, but the container often stops after backup is done, and I cannot figure out exactly why. Is it so that all non-zero error codes from restic make the container to stop? I have a few directories that are inaccessible to restic that then make the restic exit code to be non-zero most likely. 

Would be good to add some log/msg about the reason for container stopping. 

Thanks,
Vesa

**Comment by djmaze** [comment]
Hello @vsuontam and sorry for the late answer.

The container should not stop by itself since everything is run via a cron scheduler inside the container. Any chance you overrode the entrypoint or command of the container? Can you post some (part of) logs so we can see what happened?

**Comment by vsuontam** [comment]
Hello @djmaze, no problem. 

I like the resticker a lot, using it with my Synology device. It is found in the docker image registry via the GUI there. (The only problem there is that it doesn't display the version number correctly, but just always displays "latest" even though it is not the latest).

I did notice that the container was exiting because of the non-zero exit value for restic as I suspected: I had some unreadable directories in my backup, and once I changed the permissions on this, this didn't happen.
Restic returns "3" in this error (if some source files could not be read, Restic does return "1" on fatal errors), and I would love if there was an option to not have the container stopping on this "3" error. Something like: RESTICKER_DONT_EXIT_ON_NON_READABLE_FILES. Or at least add a debug line when exiting the container, as it was slightly unexpected.

**Reply by ThomDietrich** [reply]
I would vote that the container should be stopped (so the health of the container is clearly not good, from an outside perspective). I agree that the error should definitely be logged explicitly so the user understands the cause of the issue.

**Reply by djmaze** [reply]
I still don't get why the container stops for @vsuontam. This should not happen and I need more details about the configuration to debug this.

Changing the behaviour so that the container is stopped on fatal restic errors would be a breaking change in behaviour, which would necessitate a major version bump. People would need to adjust their configuration, and I am not sure it would really be an improvement. (I am open for arguments on this one though.)
